### PR TITLE
[SPARK-15542][SparkR] Make error message clear for script './R/install-dev.sh' when R is missing on Mac

### PR DESCRIPTION
**Pull request by keypointt**
https://issues.apache.org/jira/browse/SPARK-15542
## What changes were proposed in this pull request?

When running`./R/install-dev.sh` in **Mac OS EI Captain** environment, I got

```
mbp185-xr:spark xin$ ./R/install-dev.sh
usage: dirname path
```

This message is very confusing to me, and then I found R is not properly configured on my Mac when this script is using `$(which R)` to get R home.

I tried similar situation on CentOS with R missing, and it's giving me very clear error message while MacOS is not.
on CentOS:

```
[root@ip-xxx-31-9-xx spark]# which R
/usr/bin/which: no R in (/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib/jvm/java-1.7.0-openjdk.x86_64/bin:/root/bin)
```

but on Mac, if not found then nothing returned and this is causing the confusing message for R build failure and running R/install-dev.sh:

```
mbp185-xr:spark xin$ which R
mbp185-xr:spark xin$
```

Here I just added a clear message for this miss configuration for R when running `R/install-dev.sh`.

```
mbp185-xr:spark xin$ ./R/install-dev.sh
Cannot find R home by running 'which R', please make sure R is properly installed.
```
## How was this patch tested?

Manually tested on local machine.

### CODE REVIEW

**Comment by keypointt** [comment]
Hi @jkbradley do you mind have a look at this one? thanks a lot

File: R/install-dev.sh [hunk 1]
```diff
@@ -38,7 +38,12 @@ pushd $FWDIR > /dev/null
 if [ ! -z "$R_HOME" ]
   then
     R_SCRIPT_PATH="$R_HOME/bin"
-   else
+  else
+    # if command 'which R' finds no R home, then exit
+    if ! which R >/dev/null; then
```
Comment (srowen) [inline R/install-dev.sh]: I think "command" is the preferred way to do this. See how java is detected in `spark-class`.

File: R/install-dev.sh [hunk 2]
```diff
@@ -38,7 +38,12 @@ pushd $FWDIR > /dev/null
 if [ ! -z "$R_HOME" ]
   then
     R_SCRIPT_PATH="$R_HOME/bin"
-   else
+  else
+    # if system wide R_HOME is not found, then exit
+    if ! [ `command -v R` ]; then
```
Comment (srowen) [inline R/install-dev.sh]: Is it not `[ !` instead of `! [` -- were you able to test this? I doubt the integration tests have a path where R doesn't exist.

Comment (keypointt) [inline R/install-dev.sh]: I just tried on my local machine (I removed R installation), and either way is working, giving me the same message
```
RENs-MacBook-Pro:spark xin$ ./R/install-dev.sh
Cannot find 'R_HOME'. Please specify 'R_HOME' or make sure R is properly installed.
```

if `[ ! ` is a better practise I can change to it, I'm just know very basic bash :P

Comment (srowen) [inline R/install-dev.sh]: CC @nchammas but I have only seen `[ !`

Comment (nchammas) [inline R/install-dev.sh]: Yeah, we typically put the `!` after the test:
* https://github.com/apache/spark/blob/6d506c9ae9a2519d1a93e788ae5316d4f942d35d/dev/lint-python#L44
* https://github.com/apache/spark/blob/6d506c9ae9a2519d1a93e788ae5316d4f942d35d/dev/lint-java#L25

(In Bash, `[ ... ]` and `test` are synonyms.)

Comment (keypointt) [inline R/install-dev.sh]: Thanks a lot for this, I've corrected it to be `[ !`

**Comment by keypointt** [comment]
Hi @srowen is there anything extra I should do?

**Comment by srowen** [comment]
Merged to master/2.0

**Comment by keypointt** [comment]
Thank you Sean :)

**Pull request closed without merging**